src/lib/utils/plugin_manager.cpp

```diff
@@ -74,16 +74,25 @@ void PluginManager::load_plugin(const std::filesystem::path& path) {
   plugin_handle_wrapper.plugin->start();
   _plugins[plugin_name] = std::move(plugin_handle_wrapper);
 
-  // Add the newly loaded plugin's user executable functions to our map of functions.
-  const auto& user_executable_functions = _plugins[plugin_name].plugin->provided_user_executable_functions();
+  // Add the newly loaded plugin's user executable function and benchmark hooks to our maps of functions.
+  auto& plugin_ref = *_plugins[plugin_name].plugin;
+  const auto& user_executable_functions = plugin_ref.provided_user_executable_functions();
   for (const auto& [function_name, function_pointer] : user_executable_functions) {
     _user_executable_functions[{plugin_name, function_name}] = function_pointer;
   }
+  const auto& pre_benchmark_hook = plugin_ref.pre_benchmark_hook();
+  if (pre_benchmark_hook) {
+    _pre_benchmark_hooks[plugin_name] = *pre_benchmark_hook;
+  }
+  const auto& post_benchmark_hook = plugin_ref.post_benchmark_hook();
+  if (post_benchmark_hook) {
+    _post_benchmark_hooks[plugin_name] = *post_benchmark_hook;
+  }
 }
 
 void PluginManager::exec_user_function(const PluginName& plugin_name, const PluginFunctionName& function_name) {
   Assert(_user_executable_functions.contains({plugin_name, function_name}),
-         "There is no " + function_name + " defined for plugin " + plugin_name + ".");
+         "There is no " + function_name + " defined for plugin `" + plugin_name + ".");
```

Comment:
Either remove or change to straight quotes with an end quote as well.

Reply:
Argh, copied it from the here (and obviously not in the correct way.
https://github.com/hyrise/hyrise/blob/54770d02e097b783bde1c1924f45cca43502a4f3/src/lib/utils/plugin_manager.cpp#L94
I'd go for `'` if you agree - just like emphasizing the user-provided names in the messages.